components/mbedtls_cxx/mbedtls_wrap.cpp

```diff
@@ -1,23 +1,36 @@
 /*
- * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
  *
  * SPDX-License-Identifier: Apache-2.0
  */
+#include <mbedtls/timing.h>
 #include "mbedtls/ctr_drbg.h"
 #include "mbedtls/ssl.h"
 #include "mbedtls_wrap.hpp"
 
-bool Tls::init(is_server server, do_verify verify)
+bool Tls::init(is_server server, do_verify verify, TlsConfig *config)
 {
     const char pers[] = "mbedtls_wrapper";
+    if (config == nullptr) {
+
+    }
+    is_server_ = server == is_server{true};
+    is_dtls_ = config ? config->is_dtls : false;
+    uint32_t timeout = config ? config->timeout : 0;
+    const_buf client_id = config ? config->client_id : const_buf{};
     mbedtls_entropy_init(&entropy_);
     mbedtls_ctr_drbg_seed(&ctr_drbg_, mbedtls_entropy_func, &entropy_, (const unsigned char *)pers, sizeof(pers));
-    int ret = mbedtls_ssl_config_defaults(&conf_, server == is_server{true} ? MBEDTLS_SSL_IS_SERVER : MBEDTLS_SSL_IS_CLIENT, MBEDTLS_SSL_TRANSPORT_STREAM, MBEDTLS_SSL_PRESET_DEFAULT);
```

Comment:
Need more clarifications between `conf_` and `config`

Reply:
I'll add a description